src/api/profile/profile.api.js

```diff
@@ -16,16 +15,60 @@ export class ProfileError {
     }
 }
 
+/**
+ * @typedef ElevationProfilePoint
+ * @property {Number} dist Distance from first to current point (relative to the whole profile, not
+ *   by segments)
+ * @property {[Number, Number]} coordinate Coordinate of this point in the current projection
+ * @property {Number | null} elevation In the COMB elevation model
+ * @property {Boolean} hasElevationData True if some elevation data are present
+ */
+
+/**
+ * How many coordinate we will let a chunk have before splitting it into multiple chunks
+ *
+ * Backend has a hard limit at 5k, we take a conservative approach with 3k.
+ *
+ * @type {number}
+ */
+const MAX_CHUNK_LENGTH = 3000
+
+/**
+ * @param {CoordinatesChunk | null} [chunks]
+ * @returns {null | CoordinatesChunk[]}
+ */
+function splitIfTooManyPoints(chunks = null) {
```

Comment:
It seems to be working well, but I would add a small unit test to ensure this stays so.